validator-api/src/coconut/mod.rs

```diff
@@ -0,0 +1,71 @@
+use crate::Config;
+use coconut_interface::{BlindSignRequestBody, BlindedSignatureResponse, VerificationKeyResponse};
+use coconut_rs::{elgamal::PublicKey, Attribute, BlindSignRequest, BlindedSignature, Parameters};
+use getset::{CopyGetters, Getters};
+use rocket::serde::json::Json;
+use rocket::State;
+
+#[derive(Getters, CopyGetters, Debug)]
+pub struct InternalSignRequest {
+    // Total number of paraeters to generate for
+    #[getset(get_copy)]
+    total_params: u32,
+    #[getset(get)]
+    public_attributes: Vec<Attribute>,
+    #[getset(get)]
+    public_key: PublicKey,
+    #[getset(get)]
+    blind_sign_request: BlindSignRequest,
+}
+
+impl InternalSignRequest {
+    pub fn new(
+        total_params: u32,
+        public_attributes: Vec<Attribute>,
+        public_key: PublicKey,
+        blind_sign_request: BlindSignRequest,
+    ) -> InternalSignRequest {
+        InternalSignRequest {
+            total_params,
+            public_attributes,
+            public_key,
+            blind_sign_request,
+        }
+    }
+}
+
+pub fn blind_sign(request: InternalSignRequest, config: &Config) -> BlindedSignature {
```

Comment:
Does it really make sense for "sign" function to take entire config as an argument? I think it should either take secret key or there should be some method defined on the secret key itself.